src/win32/posix_w32.c

```diff
@@ -981,3 +981,55 @@ int p_inet_pton(int af, const char *src, void *dst)
 	errno = EINVAL;
 	return -1;
 }
+
+ssize_t p_pread(git_file fd, void *data, size_t size, git_off_t offset)
+{
+	HANDLE fh;
+	DWORD rsize = 0;
+	OVERLAPPED ov = {0};
+	LARGE_INTEGER pos = {0};
+
+	/* TODO: Loop over to read in smaller chunks to support large reads */
+	if (!git__is_uint32(size)) {
+		errno = E2BIG;
```

Comment:
maybe

```suggestion
		errno = EINVAL;
```

? `E2BIG` is used for argument lists, like in [`execve(2)`](https://man7.org/linux/man-pages/man2/execve.2.html):

>        E2BIG  The total number of bytes in the environment (envp) and
>              argument list (argv) is too large.